packages/agreement-process/src/services/agreementContractBuilder.ts

```diff
@@ -1,35 +1,348 @@
+};
+
+const getPdfPayload = async (
+  selfcareId: SelfcareId,
+  agreement: Agreement,
+  eservice: EService,
+  consumer: Tenant,
+  producer: Tenant,
+  seed: UpdateAgreementSeed,
+  attributeQuery: AttributeQuery
+): Promise<AgreementContractPDFPayload> => {
+  const getTenantText = (name: string, origin: string, value: string): string =>
+    origin === "IPA" ? `"${name} (codice IPA: ${value})` : name;
+
+  const getCertifiedAttributeHtml = (
+    certifiedAttributes: Array<[Attribute, CertifiedTenantAttribute]>
+  ): string =>
+    certifiedAttributes
+      .map(
+        (attTuple: [Attribute, CertifiedTenantAttribute]) => `
+        <div>
+          In data <strong>${dateAtRomeZone(
+            attTuple[1].assignmentTimestamp
+          )}</strong> alle ore <strong>${timeAtRomeZone(
+          attTuple[1].assignmentTimestamp
+        )}</strong>,l’Infrastruttura ha registrato il possesso da parte del Fruitore del seguente attributo <strong>${
+          attTuple[0].name
+        }</strong> certificato,necessario a soddisfare il requisito di fruizione stabilito dall’Erogatore per l’accesso all’E-service.
+        </div>`
+      )
+      .join("");
+
+  const getDeclaredAttributeHtml = (
+    declaredAttributes: Array<[Attribute, DeclaredTenantAttribute]>
+  ): string =>
+    declaredAttributes
+      .map(
+        (attTuple: [Attribute, DeclaredTenantAttribute]) => `
+      <div>
+         In data <strong>${dateAtRomeZone(
+           attTuple[1].assignmentTimestamp
+         )}</strong> alle ore <strong>${timeAtRomeZone(
+          attTuple[1].assignmentTimestamp
+        )}</strong>,
+         l’Infrastruttura ha registrato la dichiarazione del Fruitore di possedere il seguente attributo <strong>${
+           attTuple[0].name
+         }</strong> dichiarato
+         ed avente il seguente periodo di validità ________,
+         necessario a soddisfare il requisito di fruizione stabilito dall’Erogatore per l’accesso all’E-service.
+      </div>`
+      )
+      .join("");
+
+  const getVerifiedAttributeHtml = (
+    verifiedAttributes: Array<[Attribute, VerifiedTenantAttribute]>
+  ): string =>
+    verifiedAttributes
+      .map(
+        (attTuple: [Attribute, VerifiedTenantAttribute]) => `
+        <div>
+          In data <strong>${dateAtRomeZone(
+            attTuple[1].assignmentTimestamp
+          )}</strong> alle ore <strong>${timeAtRomeZone(
+          attTuple[1].assignmentTimestamp
+        )}</strong>,
+          l’Infrastruttura ha registrato la dichiarazione del Fruitore di possedere il seguente attributo <strong>${
+            attTuple[0].name
+          }</strong>,
+          verificata dall’aderente ________ OPPURE dall’Erogatore stesso in data <strong>${dateAtRomeZone(
+            attTuple[1].assignmentTimestamp
+          )}</strong>,
+          necessario a soddisfare il requisito di fruizione stabilito dall’Erogatore per l’accesso all’E-service.
+        </div>`
+      )
+      .join();
```

Comment:
Could these be three dedicated template files that you can interpolate here? What is the reason for hardcoding the HTML here and not having template files also for this part?

Reply:
Good suggestion, they could be a isolated template but for performance reasons I prefer to avoid a new read operation from file system

Reply:
I also prefer to avoid to embed the html here

Reply:
@MalpenZibo as discuss offline I prefer to handle this improvement in a second moment, I track this work in https://pagopa.atlassian.net/browse/IMN-572?atlOrigin=eyJpIjoiZTRkMDI2NGNiNzA1NGE3NmE5MzM4ZGU2NTU2ODdiZWMiLCJwIjoiaiJ9